packages/back-end/src/queryRunners/AutomaticDimensionQueryRunner.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  ExperimentDimension,
+  AutomaticDimensionInterface,
+  AutomaticDimensionQueryResponseRows,
+  AutomaticDimensionResult,
+} from "../types/Integration";
+import { Queries, QueryStatus } from "../../types/query";
+import {
+  getAutomaticDimensionById,
+  updateAutomaticDimension,
+} from "../models/AutomaticDimensionModel";
+import { QueryRunner, QueryMap } from "./QueryRunner";
+
+export type AutomaticDimensionParams = {
+  exposureQueryId: string;
+  lookbackDays: number;
+};
+
+export class AutomaticDimensionQueryRunner extends QueryRunner<
+  AutomaticDimensionInterface,
+  AutomaticDimensionParams,
+  AutomaticDimensionResult[]
+> {
+  async startQueries(params: AutomaticDimensionParams): Promise<Queries> {
+    const exposureQuery = (
+      this.integration.settings?.queries?.exposure || []
+    ).find((q) => q.id === params.exposureQueryId);
+
+    const dimensions: ExperimentDimension[] = (
+      exposureQuery?.dimensions || []
+    ).map((id) => ({
+      type: "experiment",
+      id,
+    }));
+
+    return [
+      await this.startQuery({
+        name: "automaticdimensions",
+        query: this.integration.getAutomaticDimensionQuery({
+          exposureQueryId: params.exposureQueryId,
+          dimensions: dimensions,
+          lookbackDays: params.lookbackDays,
+        }),
+        dependencies: [],
+        run: (query, setExternalId) =>
+          this.integration.runAutomaticDimensionQuery(query, setExternalId),
+        process: (rows) => rows,
+      }),
+    ];
+  }
+  async runAnalysis(queryMap: QueryMap): Promise<AutomaticDimensionResult[]> {
+    const automaticDimension = queryMap.get("automaticdimensions")
+      ?.result as AutomaticDimensionQueryResponseRows;
+
+    // Group by experiment and exposureQuery
+    const dimValueMap = new Map<string, { name: string; percent: number }[]>();
+    automaticDimension.forEach((d) => {
+      const dimName = d.dimension_name.replace("dim_exp_", "");
```

Comment:
Probably want to use regex to make sure you're only replacing this at the start of the string